### PR TITLE
Add swagger for above subscription scope: billing account and department

### PR DESCRIPTION
**Pull request by icodetolive**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 1]
```diff
@@ -114,6 +114,84 @@
         }
       }
     },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.CostManagement/reports": {
+      "get": {
+        "tags": [
+          "Reports"
+        ],
+        "operationId": "Report_ListByBillingAccount",
```
Comment (hovsepm) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: >Report_ [](start = 24, length = 7)

should be `Reports` - plural

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 2]
```diff
@@ -114,6 +114,84 @@
...
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/departments/{departmentId}/providers/Microsoft.CostManagement/reports": {
+      "get": {
+        "tags": [
+          "Reports"
+        ],
+        "operationId": "Report_ListByDepartment",
```
Comment (hovsepm) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: >Report_ListByDepartment [](start = 24, length = 23)

should be `Reports` - plural

**Comment by hovsepm** [comment]
@ravbhatnagar please take a look. This PR adds bunch of a new endpoints in the existing API.

**Comment by hovsepm** [comment]
@icodetolive please fix Model validation errors https://travis-ci.org/Azure/azure-rest-api-specs/jobs/422824891

**Comment by icodetolive** [comment]
@hovsepm  fixed the model validations

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 3]
```diff
@@ -391,6 +469,274 @@
         }
       }
     },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.CostManagement/reports/{reportName}": {
+      "get": {
+        "tags": [
+          "Reports"
+        ],
+        "operationId": "Report_GetByBillingAccount",
```
Comment (hovsepm) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: >Report_ [](start = 24, length = 7)

should be Reports

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 4]
```diff
@@ -391,6 +469,274 @@
...
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "Reports"
+        ],
+        "operationId": "Report_CreateOrUpdateByBillingAccount",
```
Comment (hovsepm) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: >Report_CreateOrUpdateByBillingAccount [](start = 24, length = 37)

should be Reports

**Comment by hovsepm** [comment]
@icodetolive please fix everywhere ` "operationId": "Report_` to plural ` "operationId": "Reports_`

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json (hunk 3 above)
Comment (icodetolive) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: Done

**Review by hovsepm — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Pull request merged**